provider/datastore/keystore.go

```diff
@@ -218,10 +237,10 @@ func (s *KeyStore) ResetCids(ctx context.Context, keysChan <-chan cid.Cid) error
 	if err != nil {
 		return fmt.Errorf("KeyStore empty failed during reset: %w", err)
 	}
-	keys := make([]mh.Multihash, s.gcBatchSize)
+	keys := make([]mh.Multihash, 0, s.gcBatchSize)
 	i := 0
 	for c := range keysChan {
-		keys[i] = c.Hash()
+		keys = append(keys, c.Hash())
 		i++
 		if i == s.gcBatchSize {
```

Comment:
You can remove the variable `i` and just use `len(keys)`
```suggestion
		keys = append(keys, c.Hash())
		if len(keys) == s.gcBatchSize {
```